pkg/nats/proxy/compute_handler.go

```diff
@@ -126,21 +128,31 @@ func sendResponse[Response any](conn *nats.Conn, reply string, result *concurren
 	return conn.Publish(reply, resultData)
 }
 
-func processAndStream[Request, Response any](ctx context.Context, streamingClient *stream.Client, msg *nats.Msg,
+func processAndStream[Request, Response any](ctx context.Context, streamingClient *stream.ProducerClient, msg *nats.Msg,
 	f handlerWithResponse[Request, <-chan *concurrency.AsyncResult[Response]]) {
 	if msg.Reply == "" {
 		log.Ctx(ctx).Error().Msgf("streaming request on %s has no reply subject", msg.Subject)
 		return
 	}
+
 	writer := streamingClient.NewWriter(msg.Reply)
+	streamRequest := new(stream.Request)
+	err := json.Unmarshal(msg.Data, streamRequest)
+	if err != nil {
+		_ = writer.CloseWithCode(stream.CloseBadRequest,
+			fmt.Sprintf("error decoding %s: %s", reflect.TypeOf(streamRequest).Name(), err))
+		return
+	}
+
 	request := new(Request)
-	err := json.Unmarshal(msg.Data, request)
+	err = json.Unmarshal(streamRequest.Data, request)
 	if err != nil {
 		_ = writer.CloseWithCode(stream.CloseBadRequest,
 			fmt.Sprintf("error decoding %s: %s", reflect.TypeOf(request).Name(), err))
 		return
 	}
 
+	streamingClient.AddConnDetails(ctx, &streamRequest.ConnectionDetails)
 	ch, err := f(ctx, *request)
 	if err != nil {
 		_ = writer.CloseWithCode(stream.CloseInternalServerErr,
```

Comment:
I think we have another problem (not related to this epic) where we don't close the logging stream from the source (e.g. docker container) when the producer closes the writer. Not to be addressed as part of this epic, but worth verifying and opening a separate issue to follow up.

Reply:
Should it be okay to create an epic for it ?

Reply:
https://github.com/bacalhau-project/bacalhau/issues/4040